hexa/datasets/models.py

```diff
@@ -144,6 +145,12 @@ def create_if_has_perm(
             description=description,
         )
 
+        if isinstance(principal, PipelineRunUser):
+            pipeline_run_version = PipelineRunDatasetVersion(
```

Comment:
`PipelineRunDatasetVersion.objects.create` would directly create the object instead of these two lines